specification/cognitiveservices/data-plane/CustomVision/Training/stable/v3.0/Training.json

```diff
@@ -0,0 +1,4542 @@
+            "required": true,
+            "type": "string",
+            "format": "uuid",
+            "x-nullable": false
+          },
+          {
+            "name": "updatedProject",
+            "in": "body",
+            "description": "The updated project model.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/Project"
+            }
+          },
+          {
+            "$ref": "#/parameters/ApiKey"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/Project"
+            }
+          },
+          "default": {
+            "description": "Error response",
+            "schema": {
+              "$ref": "#/definitions/CustomVisionError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Successful UpdateProject request": {
+            "$ref": "./examples/UpdateProject.json"
+          }
+        }
+      }
+    },
+    "/projects/{projectId}/train": {
+      "post": {
+        "tags": [
+          "ProjectApi"
+        ],
+        "summary": "Queues project for training.",
+        "operationId": "TrainProject",
+        "consumes": [
+
+        ],
+        "produces": [
+          "application/json",
+          "application/xml",
+          "text/xml"
+        ],
+        "parameters": [
+          {
+            "name": "projectId",
+            "in": "path",
+            "description": "The project id.",
+            "required": true,
+            "type": "string",
+            "format": "uuid",
+            "x-nullable": false
+          },
+          {
+            "name": "trainingType",
+            "in": "query",
+            "description": "The type of training to use to train the project (default: Regular).",
+            "required": false,
+            "type": "string",
+            "enum": [
+              "Regular",
+              "Advanced"
+            ]
+          },
+          {
+            "name": "reservedBudgetInHours",
+            "in": "query",
+            "description": "The number of hours reserved as budget for training (if applicable).",
+            "required": false,
+            "type": "integer",
+            "format": "int32",
+            "default": 0
+          },
+          {
+            "name": "forceTrain",
+            "in": "query",
+            "description": "Whether to force train even if dataset and configuration does not change (default: false).",
+            "required": false,
+            "type": "boolean",
+            "default": false
+          },
+          {
+            "name": "notificationEmailAddress",
+            "in": "query",
+            "description": "The email address to send notification to when training finishes (default: null).",
+            "required": false,
+            "type": "string",
+            "x-nullable": true
```

Comment:
This probably shouldn't be nullable.

Reply:
Thanks, I removed the nullable: true on the other query fields as well. They are optional, not nullable.